src/test/java/org/apache/commons/io/output/ByteArrayOutputStreamTest.java

```diff
@@ -350,6 +353,31 @@ public void testWriteZero(final String baosName, final BAOSFactory<?> baosFactor
         }
     }
 
+    @ParameterizedTest(name = "[{index}] {0}")
+    @ValueSource(ints = {1024, 64})
+    public void testPrivateFieldOfCurrentBufferIndex(int size) throws NoSuchFieldException, IllegalAccessException {
+        // Use the default size and custom size.
+        final ByteArrayOutputStream os = new ByteArrayOutputStream(size);
+
+        // Get private field object.
+        final Field field = AbstractByteArrayOutputStream.class.getDeclaredField("currentBufferIndex");
```

Comment:
@code-ferry 
Thank you for your update .
You'll want to use Mockito for this kind of testing, it is already a dependency.

Reply:
Actually it would better to also test the contents of the stream maybe by converting it to a string. So a black box test instead of a white box test. This would insure that the actual behavior is correct, not just the internal mechanism.

Reply:
@garydgregory
In general, unit tests should be white box tests. But I puzzled with this PR. Change the position of  the variable currentBufferIndex have no effect on the results of public methods, such as write and reset, so I used black box tests to illustrate the problem.
In other words, this PR is not necessary. I will close this issue.